internal/k8s/env.go

```diff
@@ -12,33 +10,52 @@ import (
 
 type ClusterName string
 
-func ProvideKubeContext(config *api.Config) (KubeContext, error) {
-	return KubeContext(config.CurrentContext), nil
+func ProvideKubeContext(configOrError APIConfigOrError) KubeContext {
+	config := configOrError.Config
+	if config == nil {
+		return ""
+	}
+	return KubeContext(config.CurrentContext)
 }
 
-func ProvideKubeConfig(clientLoader clientcmd.ClientConfig, contextOverride KubeContextOverride) (*api.Config, error) {
+type APIConfigOrError struct {
+	Config *api.Config
+	Error  error
+}
+
+func ProvideAPIConfig(clientLoader clientcmd.ClientConfig, contextOverride KubeContextOverride, namespaceOverride NamespaceOverride) APIConfigOrError {
 	config, err := clientLoader.RawConfig()
 	if err != nil {
-		return nil, errors.Wrap(err, "Loading Kubernetes current-context")
+		return APIConfigOrError{Error: errors.Wrap(err, "Loading Kubernetes current-context")}
 	}
 
 	// NOTE(nick): The RawConfig() accessor doesn't handle overrides.
 	// The other accessors do. So we do what ClientConfig does internally, and
 	// apply the overrides ourselves.
 	if contextOverride != "" {
 		config.CurrentContext = string(contextOverride)
+	}
 
-		// If the user explicitly passed an override, validate it.
-		err := clientcmd.ConfirmUsable(config, string(contextOverride))
-		if err != nil {
-			return nil, errors.Wrap(err, "Overriding Kubernetes context")
+	if namespaceOverride != "" {
+		context, ok := config.Contexts[config.CurrentContext]
+		if ok {
+			context.Namespace = string(namespaceOverride)
 		}
 	}
 
-	return &config, nil
+	err = clientcmd.ConfirmUsable(config, config.CurrentContext)
+	if err != nil {
+		return APIConfigOrError{Error: errors.Wrap(err, "Overriding Kubernetes context")}
```

Comment:
Minor: this is always happening now, so a more generic message like "Loading Kubernetes config" might be more applicable

Reply:
done